inference.py

```diff
@@ -16,133 +16,117 @@
 
 from tqdm import tqdm
 from pathlib import Path
-
+from utils.metrics import ComputePixelMetrics
 from models.model_choice import net
 from utils import augmentation
-from utils.utils import load_from_checkpoint, get_device_ids, gpu_stats, get_key_def, list_input_images, \
-    add_metadata_from_raster_to_sample
+from utils.geoutils import vector_to_raster
+from utils.utils import load_from_checkpoint, get_device_ids, gpu_stats, get_key_def, \
+list_input_images, pad, pad_diff, ind2rgb, add_metadata_from_raster_to_sample, _window_2D
 from utils.readers import read_parameters, image_reader_as_array
 from utils.CreateDataset import MetaSegmentationDataset
 from utils.verifications import add_background_to_num_class
 from utils.visualization import vis, vis_from_batch
+from mlflow import log_params, set_tracking_uri, set_experiment, start_run, log_artifact, log_metrics 
 
 try:
     import boto3
 except ModuleNotFoundError:
     pass
-
-
-def sem_seg_inference(model,
-                      nd_array,
-                      overlay,
-                      chunk_size,
-                      num_classes,
-                      device,
-                      meta_map=None,
-                      metadata=None,
-                      output_path=Path(os.getcwd()),
-                      index=0,
-                      debug=False):
-    """Inference on images using semantic segmentation
-    Args:
-        model: model to use for inference
-        nd_array: input raster as array
-        overlay: amount of overlay to apply
-        chunk_size: size of individual chunks to be processed during inference
-        num_classes: number of different classes that may be predicted by the model
-        device: device used by pytorch (cpu ou cuda)
-        meta_map:
-        metadata:
-        output_path: path to save debug files
-        index: (int) index of array from list of images on which inference is performed
-
-        returns a numpy array of the same size (h,w) as the input image, where each value is the predicted output.
-    """
-
+colors = {0: [0, 0, 0],  # Background
+          1: [27, 120, 55],  # Vegetation
+          2: [116, 173, 209], # Hydro
+          3: [223, 194, 125],  # Roads
+          4: [150, 150, 150]}  # Buildings     
```

Comment:
why hardcode this? won't it cause bug it num_classes > 5 ?
visualization.py was implemented to always [default to a colormap defined by matplotlib](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/visualization.py#L138).

Reply:
it's hardcoded right now because the color mapping function is only efficient for smaller arrays. I haven't found a way to optimize it using matplotlib.

Reply:
Maybe use a colormap and pick the number of color associate to the number of classes. 
```
import matplotlib.pyplot as plt
import random

number_of_colors = 8

color = ["#"+''.join([random.choice('0123456789ABCDEF') for j in range(6)])
             for i in range(number_of_colors)]
print(color)
```
(random exemple)

Reply:
The problem lies in the mapping function, not the color picker. I don't have an efficient way of mapping HTML colors without Matplotlib. It is simply not efficient for large arrays.

Reply:
Ok I understand :/ hummm

Reply:
Ok. I guess we can hardcode as a temporary approach, but it would be important to handle tasks with more than 4 classes. I'll let you figure what solution fits best.